controllers/v1/solutions.js

```diff
@@ -0,0 +1,220 @@
+const config = require('../../config/config');
+const rp = require('request-promise');
+const request = require('request');
+const model = require('../../db');
+const helperFunc = require('../../helper/chart_data');
+const filesHelper = require('../../common/files_helper');
+const authService = require('../../middleware/authentication_service');
+
+/**
+   * @api {post} /dhiti/api/v1/solutions/list 
+   * Solutions List 
+   * @apiVersion 1.0.0
+   * @apiGroup Solutions
+   * @apiHeader {String} x-auth-token Authenticity token  
+   * @apiParamExample {json} Request-Body:
+* {
+  "entityId": "",
+  "entityType": "",
+  "programId": ""
+* }
+   * @apiSuccessExample {json} Success-Response:
+*     HTTP/1.1 200 OK
+*     {
+       "result": true,
+       "data": {
+           "mySolutions": [{
+               "solutionId": "",
+               "solutionName": "",
+               "type": "",
+               "scoring": ""
+           }],
+           "allSolutions": [{
+               "solutionId": "",
+               "solutionName": "",
+               "type": "",
+               "scoring": ""
+           }]
+       }
+*     }
+   * @apiUse errorBody
+   */
+
+
+//Controller for listing solution Names
+exports.list = async function (req, res) {
+
+    try {
+
+    if (!req.body.entityId || !req.body.entityType || !req.body.programId) {
+        res.status(400);
+        var response = {
+            result: false,
+            message: 'entityId,entityType,programId are required fields'
+        }
+        res.send(response);
+    }
+    else {
+
+          //get the createdBy field
+          let id = await getCreatedByField(req, res);
+
+          let myAssessmentSolutions = await mySolutionsList(req, id, filesHelper.assessment);
```

Comment:
@deepakr123 - All 4 queries can be in parallel.

Reply:
resolved